hoodie-common/src/main/java/com/uber/hoodie/common/util/ParquetUtils.java

```diff
@@ -178,4 +207,20 @@ public static BloomFilter readBloomFilterFromParquetMetadata(Configuration confi
     }
     return records;
   }
+
+  static class RecordKeysFilterFunction implements Function<String, Boolean> {
+    private final Optional<Set<String>> candidateKeys;
```

Comment:
Make candidateKeys as non-optional. The method readRowKeysFromParquet should not even construct this filter function when there is no input map to filter. This way, there is no need to call apply for each row in parquet if there is no input map.